src/chainstate/stacks/db/mod.rs

```diff
@@ -894,87 +904,51 @@ impl StacksChainState {
             );
             receipts.push(allocations_receipt);
 
-            let mut total_vesting_amount: u128 = 0;
             if let Some(get_schedules) = boot_data.get_bulk_initial_vesting_schedules.take() {
                 info!("Initializing chain with vesting schedules");
-                let mut vesting_schedule_count = 0;
+                let mut vesting_schedules_per_block: HashMap<u64, Vec<Value>> = HashMap::new();
+                let initial_vesting_schedules = get_schedules();
+                for schedule in initial_vesting_schedules {
+                    let value = Value::Tuple(
+                        TupleData::from_data(vec![
+                            (
+                                "recipient".into(),
+                                Value::Principal(schedule.address.into()),
+                            ),
+                            ("amount".into(), Value::UInt(schedule.amount.into())),
+                        ])
+                        .unwrap(),
+                    );
+                    match vesting_schedules_per_block.entry(schedule.block_height) {
```

Comment:
Minor style nit -- if you're inserting into a HashMap, you can do it more economically like so:

```
if let Some(ref mut schedules) = vesting_schedules_per_block.get_mut(&schedule.block_height) {
   schedules.push(value);
}
else {
   vesting_schedules_per_block.insert(schedule.block_height, vec![value]);
}
```

This is what we do elsewhere in the codebase.